sympy/integrals/tests/test_failing_integrals.py

```diff
@@ -114,3 +114,79 @@ def test_issue_4941():
 def test_issue_4992():
     # Nonelementary integral.  Requires hypergeometric/Meijer-G handling.
     assert not integrate(log(x) * x**(k - 1) * exp(-x) / gamma(k), (x, 0, oo)).has(Integral)
+
+@XFAIL
+def test_issue_16084():
+    assert not integrate(log(sin(x)), (x, 0, pi/2)).has(Integral)
```

Comment:
I'm not sure this first assertion is necessary. But do you have it there in case the integral is solved but in a different form?

Reply:
I think the first thing is to be able to solve the integral in any form. For the cases where a compact answer is known, there is a second assert for that answer. It can of course be argued if the issue is closed by returning a complicated expression or not, but that is my logic behind it.

See e.g. 
https://github.com/sympy/sympy/blob/d12570e481d862f8e06292e9684b97bc7f5b4821/sympy/integrals/tests/test_failing_integrals.py#L35-L40

Maybe one should actually have two different tests, one for solving it and one for solving it "correctly"? Browsing through the failing integral tests made me realize that many of the issues had been closed, but the tests were not removed from `test_failing_integrals`. I guess the reason is that someone solves a general problem and then it may have positive effects on other parts of the code as well (although that doesn't really explain why the issues were closed...).